core/src/main/java/org/apache/accumulo/core/spi/compaction/CompactionServiceId.java

```diff
@@ -28,11 +31,21 @@
 public class CompactionServiceId extends AbstractId<CompactionServiceId> {
   private static final long serialVersionUID = 1L;
 
+  static final Cache<String,CompactionServiceId> cache = Caches.getInstance()
+      .createNewBuilder(Caches.CacheName.COMPACTION_SERVICE_ID, false).weakValues().build();
```

Comment:
When looking at the [documentation](https://github.com/ben-manes/caffeine/wiki/Eviction#reference-based), it's unclear to me whether `weakValues` alone will allow for garbage collection of the Cache key *and* value. I haven't looked at all the places the `CompactionServiceId` is used, I wonder if it's possible that a reference to it is held for a long time. Another approach that I mentioned in #4316 is [time-based eviction](https://github.com/ben-manes/caffeine/wiki/Eviction#time-based).

Reply:
Created #4336 while looking at this.